src/__tests__/as-array.ts

```diff
@@ -0,0 +1,45 @@
+import test from 'ava';
+import Collection from '../index';
+import { Item, items } from './fixtures';
+
+test('with no items loaded, #viewAsArray with no passed IDs', t => {
+  const col = new Collection<Item>();
+  t.deepEqual(col.viewAsArray(), [], 'returns an empty array');
+});
+
+test('with no items loaded, #view with invalid IDs', t => {
+  const col = new Collection<Item>();
+  t.deepEqual(col.viewAsArray(['x', 'y', 'z']), [], 'returns an empty array');
+});
+
+test('with item loading errors, #view with no passed IDs', t => {
+  const col = new Collection<Item>().withListFailure('There was a problem loading the list');
+  t.deepEqual(col.viewAsArray(), [], 'returns an empty array');
+});
+
+test('with item loading errors, #view with IDs', t => {
+  const col = new Collection<Item>().withListFailure('There was a problem loading the list');
+  t.deepEqual(col.viewAsArray(['a']), [], 'returns an empty array');
+});
+
+test('with items loaded, #view with no passed IDs', t => {
+  const col = new Collection<Item>().withList('id', items);
+  t.deepEqual(col.viewAsArray(), items, 'returns all known items');
+});
+
+test('with items loaded, #view with valid IDs', t => {
```

Comment:
can you test `#view` with a mixture of valid and invalid ids?

Reply:
⚡️ 10ee611

Good call. Added that case to both `view` and `viewAsArray`.

Side note: You'll notice that it isn't exactly what you would expect, it just filters out the invalid ones. I think that's sensible for now, but we _could_ choose to return `RemoteInitial` if any of them are invalid. I think that is pretty surprising though.